src/sonic-yang-models/yang-models/sonic-route-map.yang

```diff
@@ -0,0 +1,359 @@
+                    type string;
+                    description "Route map name";
+                }
+            }
+        }
+
+        container ROUTE_MAP {
+            list ROUTE_MAP_LIST {
+                key "name stmt_name";
+
+                leaf name {
+                    type string;
+                    description "Route map name";
+                }
+
+                leaf stmt_name {
+                    type uint16 {
+                        range 1..65535;
+                    }
+                    description "Statement name, this helps to provide multiple statements 
+                                 under a particular route-map";
+                }
+
+                leaf route_operation {
+                    type srpolsets:routing-policy-action-type;
+                    description "permit/deny operation";
+                }
+
+                leaf match_interface{
+                    type route-map-intf;
+                    description "Match based on interface";
+                }
+
+                leaf match_prefix_set{
+                    type leafref {
+                        path "/srpolsets:sonic-routing-policy-sets/srpolsets:PREFIX_SET/srpolsets:PREFIX_SET_LIST/srpolsets:name";
+                    }
+                    description
+                        "Match a prefix list name that contains IPv4 prefixes.";
+                }
+
+                leaf match_ipv6_prefix_set{
+                    type leafref {
+                        path "/srpolsets:sonic-routing-policy-sets/srpolsets:PREFIX_SET/srpolsets:PREFIX_SET_LIST/srpolsets:name";
+                    }
+                    description
+                        "Match a prefix list name that contains IPv6 prefixes.";
+                }
+
+                leaf match_protocol {
+                    type string;
+                    description "Match based on IP protocols bgp, connected, ospf, ospf3 and static";
+                }
+
+                leaf match_next_hop_set {
+                    type leafref {
+                        path "/srpolsets:sonic-routing-policy-sets/srpolsets:PREFIX_SET/srpolsets:PREFIX_SET_LIST/srpolsets:name";
+                    }
+                    description "Match based on nexthop";
+                }
+
+                leaf match_src_vrf {
+                    type union {
+                        type string {
+                            pattern "default";
+                        }
+                        type leafref {
+                            path "/svrf:sonic-vrf/svrf:VRF/svrf:VRF_LIST/svrf:name";
+                        }
+                    }
+                    description "Match based on source VRF";
+                }
+
+                leaf-list match_neighbor {
+                    type union {
+                        type inet:ip-address;
+                        type leafref {
+                            path "/prt:sonic-port/prt:PORT/prt:PORT_LIST/prt:name";
```

Comment:
subinterface please

Reply:
We'll add it once the subinterfaces YANG is available, not as part of this initial PR.